src/app/core/device.ts

```diff
@@ -0,0 +1,1044 @@
+        for (let i in this.__layers) {
+            output.push(this.__layers[i].toInterchangeV1());
+        }
+        return output;
+    }
+    /**
+     * Loads layers from a JSON format into the device object
+     * @param {JSON} json 
+     * @memberof Device
+     * @returns {void}
+     */
+    __loadLayersFromJSON(json: {[index: string]: any}): void {
+        for (let i in json) {
+            let newLayer = Layer.fromJSON(json[i]);
+            this.addLayer(newLayer);
+        }
+    }
+    /**
+     * Loads feature layers from a Interchange format into the device object
+     * @param {*} json 
+     * @memberof Device
+     * @returns {void}
+     */
+    __loadFeatureLayersFromInterchangeV1(json: Array<LayerInterchangeV1>): void {
+        for (let i in json) {
+            let newLayer = Layer.fromInterchangeV1(json[i]);
+            this.addLayer(newLayer);
+        }
+    }
+
+    /**
+     * Loads the JSON Component object into the device object
+     * @param {ComponentInterchangeV1} components
+     * @memberof Device
+     * @returns {void}
+     * @private
+     */
+    __loadComponentsFromInterchangeV1(components: Array<ComponentInterchangeV1>): void {
+        let componenttoadd;
+
+        for (let i in components) {
+            componenttoadd = Component.fromInterchangeV1(components[i]);
+            this.__components.push(componenttoadd);
+        }
+    }
+    /**
+     * Loads connections to the device object
+     * @param {Connection} connections Connections to add to the device
+     * @memberof Device
+     * @returns {void}
+     */
+    __loadConnectionsFromInterchangeV1(connections: Array<ConnectionInterchangeV1>): void {
+        let connectiontoload;
+        for (let i in connections) {
+            connectiontoload = Connection.fromInterchangeV1(this, connections[i]);
+            this.__connections.push(connectiontoload);
+        }
+    }
+    /**
+     * Converts the properties of the device to JSON 
+     * @returns {JSON} Returns a JSON format with the properties of the device
+     * @memberof Device
+     */
+    toJSON(): {[index: string]: any} {
+        let output: {[index: string]: any} = {};
+        output.name = this.__name.toJSON();
+        output.params = this.__params.toJSON();
+        output.layers = this.__layersToJSON();
+        output.groups = this.__groupsToJSON();
+        output.defaults = this.__defaults;
+        return output;
+    }
+    /**
+     * Converts to Interchange V1 format
+     * @returns {Device} Returns an Device object in Interchange V1 format
+     * @memberof Device
+     */
+    toInterchangeV1(): DeviceInterchangeV1 {
+        let output: DeviceInterchangeV1 = {
+            name = this.__name,
+            params = {
+                width: this.getXSpan(),
+                length: this.getYSpan()
+            },
+        //TODO: Use this to dynamically create enough layers to scroll through
+        // output.layers = this.__layersToInterchangeV1();
```

Comment:
We need to do this by converting all the `.layers` in device

Reply:
This is already done by '__featureLayersToInterchangeV1' which is used directly below these lines in this function to generate the property 'features'

Reply:
`FeatureLayer` is different from `Layer` when its read by the Parchmint (Interchange) format parsers. This will be an issue you will address while creating the Integration Layer, please enable the `output.layers = this.__layersToInterchangeV1();`

Reply:
`__featureLayersToInterchangeV1()` uses the property `__layers`, should `__layersToInterchangeV1()` use `__textLayers`

Reply:
Ummm not exactly, both use `.__layers` at the moment when you start working on the integration layer stuff, you'll separate them more cleanly